### PR TITLE
feat: support dataset update

### LINKED ISSUE #1178: Support dataset update
I believe dataset update is already on your roadmap given the skeleton here: https://github.com/googleapis/python-aiplatform/blob/075fa3bc16ef6d5b2fa158e2750ec0a31fd44e51/google/cloud/aiplatform/datasets/dataset.py#L600-L601

Where can the community track the timeline for that API? Thank you!

### PR DESCRIPTION
**Pull request by jaycee-li**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-aiplatform/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #1178  🦕

### CODE REVIEW

File: google/cloud/aiplatform/datasets/dataset.py [hunk 1]
```diff
@@ -597,8 +598,68 @@ def export_data(self, output_dir: str) -> Sequence[str]:
 
         return export_data_response.exported_files
 
-    def update(self):
-        raise NotImplementedError("Update dataset has not been implemented yet")
+    def update(
+        self,
```
Comment (sasha-gitg) [inline google/cloud/aiplatform/datasets/dataset.py:602]: Preference to make these keyword only arguments:

```Python
def update(
  self,
  *,
  display_name,
   ...)
```

File: google/cloud/aiplatform/datasets/dataset.py [hunk 2]
```diff
@@ -597,8 +598,68 @@ def export_data(self, output_dir: str) -> Sequence[str]:
...
+            update_mask.paths.append("description")
+
+        update_dataset = gca_dataset.Dataset(
+            name=self.resource_name,
+            display_name=display_name,
+            description=description,
+            labels=labels,
+        )
+
+        self.api_client.update_dataset(
```
Comment (sasha-gitg) [inline google/cloud/aiplatform/datasets/dataset.py:656]: Preference to assign `_gca_resource` to skip the additional GET call below.

```Python
self._gca_resource = self.api_client.update_dataset(...)

return self
```

File: tests/system/aiplatform/test_dataset.py [hunk 3]
```diff
@@ -350,3 +352,27 @@ def test_export_data(self, storage_client, staging_bucket):
         blob = bucket.get_blob(prefix)
 
         assert blob  # Verify the returned GCS export path exists
+
+    def test_update_dataset(self):
```
Comment (sasha-gitg) [inline tests/system/aiplatform/test_dataset.py:356]: Preference for unit test as well.

**Review by sasha-gitg — CHANGES_REQUESTED** [review]

File: tests/system/aiplatform/test_dataset.py [hunk 4]
```diff
@@ -350,3 +352,27 @@ def test_export_data(self, storage_client, staging_bucket):
...
+                update_request_timeout=None,
+            )
+            labels.update(_TEST_DATASET_LABELS)
+
+            assert dataset.display_name == _TEST_DATASET_DISPLAY_NAME
+            assert dataset.labels == labels
+            assert dataset.gca_resource.description == _TEST_DATASET_DESCRIPTION
+
+        finally:
+            if dataset is not None:
```
Comment (sasha-gitg) [inline tests/system/aiplatform/test_dataset.py:377]: This line is superfluous. `dataset` will only exist on assignment after `ImageDataset.create` is successful.

The test should start with `dataset=None`

**Review by sasha-gitg — APPROVED** [review]

File: tests/system/aiplatform/test_dataset.py (hunk 4 above)
Comment (jaycee-li) [reply tests/system/aiplatform/test_dataset.py:377]: If the `ImageDataset.create` is successful, then we want to delete the dataset finally. But if `create()` fails, dataset is None and we shouldn't call `dataset.delete()`. So I think it's necessary to add this line.

Comment (jaycee-li) [reply tests/system/aiplatform/test_dataset.py:377]: If the `ImageDataset.create` is successful, then we want to delete the dataset finally. But if `create()` fails, dataset is None and we shouldn't call `dataset.delete()`. So I think it's necessary to add this line.

Comment (sasha-gitg) [reply tests/system/aiplatform/test_dataset.py:377]: If create fails then `dataset`  is not defined and `if dataset is not None` will throw a `NameError`. `dataset` isn't created until after the right side of the statement, ` dataset = aiplatform.ImageDataset.create()` is successful. You can validate this behavior in Python:

```Python
def f():
    raise RuntimeError('')
    
try:
    dataset = f()
except:
    if dataset is None:
        pass
```

Comment (jaycee-li) [reply tests/system/aiplatform/test_dataset.py:377]: Got it! 

But I see other system tests like `test_create_and_import_image_dataset` and `test_create_tabular_dataset` have the same pattern that checks `if dataset is not None`. Is there any reason for that?

Comment (sasha-gitg) [reply tests/system/aiplatform/test_dataset.py:377]: No, those should be eventually fixed as well.

Comment (jaycee-li) [reply tests/system/aiplatform/test_dataset.py]: Got it. Just removed this line. Thank you!

**Pull request merged**